nncf/quantization/quantize_model.py

```diff
@@ -262,16 +266,9 @@ def compress_weights(
         if group_size is None:
             group_size = 128
 
-    if backend == BackendType.TORCH:
-        from nncf.torch.quantization.quantize_model import compress_weights_impl
-
-        return compress_weights_impl(model, mode, ratio, group_size)
-    if backend == BackendType.OPENVINO:
-        from nncf.openvino.quantization.quantize_model import compress_weights_impl
-
-        return compress_weights_impl(model, mode, ratio, group_size)
-
-    raise RuntimeError(f"Unsupported type of backend: {backend}")
+    compression_algorithm = WeightCompression(mode, ratio, group_size)
+    graph = NNCFGraphFactory.create(model)
+    return compression_algorithm.apply(model, graph)
```

Comment:
The main difference, that current implementation requires the nncf_graph.
But it shouldn't induce a lot of overhead. Will provide time and memory measurements later.
cc: @alexsu52